### ISSUE TITLE
Attempt to run setup script with PHP 8.0.0 with Dec 8, 2020 snapshot reports errors

### ISSUE THREAD

**Issue by jpbion** [phpmyadmin/phpmyadmin#16526]
### Describe the bug

Attempt to run setup script with PHP 8.0.0 with December 8, 2020 snapshot reports error as listed below.

### To Reproduce

Try to run setup under PHP 8.0.0. The following error is returned if you've enabled error output with PHP:

```
Fatal error: Uncaught TypeError: preg_replace(): Argument #3 ($subject) must be of type array|string, null given in /usr/share/apache2/htdocs/my.site/abc/libraries/classes/Controllers/Setup/HomeController.php:30 

Stack trace: 
#0 /usr/share/apache2/htdocs/my.site/abc/libraries/classes/Controllers/Setup/HomeController.php(30): preg_replace() 
#1 /usr/share/apache2/htdocs/my.site/abc/setup/index.php(79): PhpMyAdmin\Controllers\Setup\HomeController->index() 
#2 {main} thrown in /usr/share/apache2/htdocs/my.site/abc/libraries/classes/Controllers/Setup/HomeController.php on line 30
```

Attempt to start the setup script by going to https://yoursitename/phpmyadmin/setup

### Expected behavior

Run the setup script without errors.

### Screenshots

Not applicable

### Server configuration

- Operating system: Linux 5.9.12
- Web server: Apache HTTPD 2.4.46
- Database version: MySQL 8.0.22
- PHP version: 8.0.0
- phpMyAdmin version: December 8, 2020 Snapshot

### Client configuration

Not relevant 

### Additional context

None

**Comment by williamdes** [comment]
Hi !
Could you change the line 29 to `        $actionDone = Core::isValid($params['action_done'], 'scalar') ? $params['action_done'] : '';` ?

```diff
diff --git a/libraries/classes/Controllers/Setup/HomeController.php b/libraries/classes/Controllers/Setup/HomeController.php
index 0439bc09f4..8fa13a0968 100644
--- a/libraries/classes/Controllers/Setup/HomeController.php
+++ b/libraries/classes/Controllers/Setup/HomeController.php
@@ -26,7 +26,7 @@ class HomeController extends AbstractController
         $pages = $this->getPages();
 
         // Handle done action info
-        $actionDone = Core::isValid($params['action_done'], 'scalar') ? $params['action_done'] : null;
+        $actionDone = Core::isValid($params['action_done'], 'scalar') ? $params['action_done'] : '';
         $actionDone = preg_replace('/[^a-z_]/', '', $actionDone);
 
         // message handling
```

**Comment by jpbion** [comment]
The single line change mentioned above made things work in setup. To get everything running, I also needed to set permissions properly in libraries/cache and delete the existing cache. But once I did that, things seemed to work with PHP 8.0.0.

**Comment by williamdes** [comment]
> The single line change mentioned above made things work in setup. To get everything running, I also needed to set permissions properly in libraries/cache and delete the existing cache. But once I did that, things seemed to work with PHP 8.0.0.

Thank you for the feedback, I pushed b826234ebb5da52afd940fa0a5d552a1b9a3b0dd to make the fix you confirmed be part of 5.1
